src/features/network/pages/network-page.tsx

```diff
@@ -11,28 +11,26 @@ type Props = {
 const wildcardNetworkRoute = '_'
 
 export function NetworkPage({ children }: Props) {
-  const { networkId } = useRequiredParam(UrlParams.NetworkId)
+  const { networkId: currentNetworkId } = useRequiredParam(UrlParams.NetworkId)
   const networkConfigs = useNetworkConfigs()
 
-  if (!(networkId in networkConfigs) && networkId !== wildcardNetworkRoute) {
-    throw new Error(`"${networkId}" is not a valid network.`)
+  if (!(currentNetworkId in networkConfigs) && currentNetworkId !== wildcardNetworkRoute) {
+    throw new Error(`"${currentNetworkId}" is not a valid network.`)
   }
 
   const navigate = useNavigate()
-  const [selectedNetwork] = useSelectedNetwork()
-  const { pathname, search, hash } = useLocation()
+  const [selectedNetwork, setSelectedNetwork] = useSelectedNetwork()
+  const { pathname } = useLocation()
 
   useEffect(() => {
-    if (networkId === wildcardNetworkRoute) {
-      // Handle the wildcard network route
-      navigate(pathname.replace(wildcardNetworkRoute, selectedNetwork) + search + hash, { replace: true })
-    } else if (networkId !== selectedNetwork) {
-      // When a user changes the network, their history will contain routes for the previous network.
-      // This handles navigating the user to the explore page for the selected network, so they don't get 404s when navigating through history.
-      navigate(Urls.Network.Explore.build({ networkId: selectedNetwork }), { replace: true })
+    if (currentNetworkId === selectedNetwork) {
```

Comment:
Previously we had a behaviour which allowed users to hit a route with and `_` as the network and it'd select the last selected network. This no longer works.

Reply:
I added this functionality back in Out of curiosity, why was this added?

Reply:
I honestly don't remember the exact reason. We can check if it's being used in any of the AlgoKit tooling links and potentially remove.